datafusion/src/physical_plan/functions.rs

```diff
@@ -1223,18 +1231,12 @@ pub fn create_physical_expr(
         // These don't need args and input schema
         _ => create_physical_fun(fun, ctx_state)?,
     };
-    let args = coerce(args, input_schema, &signature(fun))?;
-
-    let arg_types = args
-        .iter()
-        .map(|e| e.data_type(input_schema))
-        .collect::<Result<Vec<_>>>()?;
 
     Ok(Arc::new(ScalarFunctionExpr::new(
         &format!("{}", fun),
         fun_expr,
         args,
-        &return_type(fun, &arg_types)?,
+        &data_type,
```

Comment:
We can still directly use `return_type(fun, &arg_types)?`, don't need declare a variable name `data_type`. Because only use it here.